### PR TITLE
Dbf xbrl mapping

### PR DESCRIPTION
**Pull request by zaneselvans**
## What this PR does
* Adds some pre-commit / git magic to ensure that all CSVs (and other text files) have Unix-style line endings in the repository, so we don't get full-file diffs whenever we edit a CSV using Excel / Libre Office.
* Applies that magic to all our existing CSVs.
* Renames some of the values & columns in the DBF to XBRL row mapping CSV to be more generic, since the nomenclature of the Plant in Service table doesn't apply across all tables we are using it to transform.
* Add a new CSV `dbf_to_xbrl_tables.csv` that associates old DBF tables with new XBRL tables, so we know which of them are related to each other. This will supplant the table names dictionary we have in `pudl.extract.ferc1` right now, and is necessary because there are one-to-many and many-to-one relationships between these tables.
* Corrected & added details to our FERC Form 1 data dictionary table descriptions -- stuff I discovered while trying to map the DBF & XBRL tables to each other.
* Removed the old row-mapping metadata for the `plant_in_service_ferc1` table, since it has been replaced by the new DBF + XBRL alignment process & mapping.

## PR Checklist

Before requesting a review of your pull request, please make sure you've done the
following:

* [x] Merge the most recent version of `dev` (or the appropriate upstream branch) into
      your branch and resolved any merge conflicts. You may need to do this several
      times over the course of a PR as `dev` changes frequently.
* [x] Verify that all of the CI checks on your PR are passing. See
      [Running Tests with Tox](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
      for details on how to run the full test suite locally if you need to debug a
      particular failure.
* [x] Ensure that the docstrings for any new modules, classes, functions, or methods are
      descriptive enough for developers and users to understand your code.
* [x] If you expanded data coverage or changed the outputs, ensure that the full
      [data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
      pass locally on a fresh DB.
* [x] If you've added new functions or classes, ensure that they have at least basic
      unit tests.
* [x] If you've added new analyses, make sure they include defensive sanity checks that
      will catch unexpected data issues.
* [x] Update the
      [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html)
      to reflect your changes. Make sure to reference the PR and any related issues.
* [x] Do your own review of the PR. Add comments highlighting areas where you have
      questions you'd like reviewers to answer, known issues, solutions you're
      unsatisfied with, or other things that deserve special attention from the
      reviewer.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +1,12 @@
...
+*.csv text
+*.py text
+*.json text
+*.yml text
+*.yaml text
+*.rst text
+*.toml text
+*.ini text
+*.txt text
+*.md text
```
Comment (zaneselvans) [inline .gitattributes:12]: Are there any other "text" file endings that I'm forgetting here?

Comment (zaneselvans) [reply .gitattributes:12]: This is a git-specific way of ensuring all of the line endings are standardized to LF (the unix standard) inside the repository, which means that when we edit CSV files with Excel / Libre Office only the lines we change will show up as changed in Git.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -22,7 +22,8 @@ repos:
       - id: check-case-conflict # Avoid case sensitivity in file names.
       - id: debug-statements # Watch for lingering debugger calls.
       - id: end-of-file-fixer # Ensure there's a newline at EOF.
-      - id: mixed-line-ending # Only newlines, no line-feeds.
+      - id: mixed-line-ending # Use Unix line-endings to avoid big no-op CSV diffs.
+        args: ["--fix=lf"]
```
Comment (zaneselvans) [inline .pre-commit-config.yaml:26]: This is another way of ensuring that all of the text files have standard line endings. Probably don't need both the git and the pre-commit hook versions but they also don't seem to conflict with each other, and the pre-commit hook will be forcibly applied by pre-commit-ci as a backup.

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv [hunk 3]
```diff
@@ -1873,62 +1873,62 @@ f1_elc_oper_rev_nb,1,1994,Includes $ unbilled revenues & MWH relating to unbille
...
-f1_elctrc_erg_acct,9,1994,Net Generation (Enter Total of lines 3 through 8),total,net_energy_generation,
-f1_elctrc_erg_acct,10,1994,Purchases,,megawatt_hours_purchased_dbfonly,"Split into 2 megawatt_hours_purchased_ fields in XBRL, which should probably be summed to produce this field."
+f1_elctrc_erg_acct,3,1994,Steam,reported_value,steam_generation,
+f1_elctrc_erg_acct,4,1994,Nuclear,reported_value,nuclear_generation,
+f1_elctrc_erg_acct,5,1994,Hydro-Conventional,reported_value,hydro_conventional_generation,
+f1_elctrc_erg_acct,6,1994,Hydro-Pumped Storage,reported_value,hydro_pumped_storage_generation,
+f1_elctrc_erg_acct,7,1994,Other,reported_value,other_energy_generation,
+f1_elctrc_erg_acct,8,1994,Less Energy for Pumping,reported_value,pumping_energy,
+f1_elctrc_erg_acct,9,1994,Net Generation (Enter Total of lines 3 through 8),calculated_value,net_energy_generation,
+f1_elctrc_erg_acct,10,1994,Purchases,reported_value,megawatt_hours_purchased_dbfonly,"Split into 2 megawatt_hours_purchased_ fields in XBRL, which should probably be summed to produce this field."
```
Comment (cmgosnell) [inline src/pudl/package_data/ferc1/dbf_to_xbrl.csv:1883]: i like the rename of these row types.... but does anything need to be changed downstream to accommodate that with that?

**Review by cmgosnell — APPROVED** [review]
this looks mostly good. this is a future suggestion, but I think for this kind of pr it would have been helpful to have a lil list of the things you did and why bc there are a few disparate things going on in here. but generally...
- lots of csv's touched bc of the gitattributes/pre-commit changes
- more table descriptions in our ferc db data dictionaries
- renaming `xbrl_column_stem` to `xbrl_factoid`
- renaming some of the `row_types` in the dbf to xbrl map.
- anything else?

I have a few calrifying questions but generally this looks just fine to me.

File: src/pudl/package_data/ferc1/dbf_to_xbrl_tables.csv [hunk 4]
```diff
@@ -0,0 +1,181 @@
+dbf_table,xbrl_table
+f1_106_2009,information_on_formula_rates_106
+f1_106a_2009,information_on_formula_rates_106a
```
[the remark this answers is not in the document]
Comment (zaneselvans) [reply src/pudl/package_data/ferc1/dbf_to_xbrl_tables.csv:3]: This file (`dbf_to_xbrl_tables.csv`) is an association table which will replace the simple one-to-one dictionary mapping that tells us which DBF + XBRL tables need to be extracted together. It's not being used programmatically anywhere yet, but we'll need it when we get to the many-to-one or one-to-many tables, and I've been using it in a notebook to read related tables for planning their mapping transforms.

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv (hunk 3 above)
Comment (zaneselvans) [reply src/pudl/package_data/ferc1/dbf_to_xbrl.csv:1883]: It turned out that these values were not yet being referenced anywhere, so I didn't need to change anything downstream. For the moment they are just serving to document the table. I think they'll become more useful when we get into the aggregation processes.

**Comment by zaneselvans** [comment]
@cmgosnell I added a description to the PR. Sorry I left that out. This is a mishmash.

**Pull request merged**